### PR TITLE
kTfLiteNullBufferHandle must be macro

### PR DESCRIPTION
**Pull request by mattn**
tensorflow/lite/c/c_api_internal.h have constant variable kTfLiteNullBufferHandle (not macro or static const). So we can't include this header file in several codes.

### CODE REVIEW

**Comment by haozha111** [comment]
I think there are current many .cc files including the c_api_internal.h. What's the specific error your encounter?

**Comment by mattn** [comment]
a.c
```
#include <tensorflow/lite/experimental/c/c_api.h>

extern void foo();

int
main() {
  foo();
}
```
b.c
```
#include <tensorflow/lite/experimental/c/c_api.h>

void
foo() {
}
```
```
$ gcc -I/path/to/tensorflow/lite//experimental/c -c a.c
$ gcc -I/path/to/tensorflow/lite//experimental/c -c b.c
$ gcc -o tmp.exe a.o b.o 
b.o:b.c:(.rdata+0x0): multiple definition of `kTfLiteNullBufferHandle'
a.o:a.c:(.rdata+0x0): first defined here
```

File: tensorflow/lite/c/c_api_internal.h [hunk 1]
```diff
@@ -279,7 +279,7 @@ typedef enum {
 // The delegates should use zero or positive integers to represent handles.
 // -1 is reserved from unallocated status.
 typedef int TfLiteBufferHandle;
-const TfLiteBufferHandle kTfLiteNullBufferHandle = -1;
+#define kTfLiteNullBufferHandle ((TfLiteBufferHandle)(-1))
```
Comment (haozha111) [inline tensorflow/lite/c/c_api_internal.h]: how about just kTfLiteNullBufferHandle (-1)

Seems like the type conversion is not necessary here.

Comment (mattn) [reply tensorflow/lite/c/c_api_internal.h]: I'm okay. will fix in later. (But I can't find necessity of `TfLiteBufferHandle` now).

**Comment by mattn** [comment]
oops. sorry. will fix

**Comment by mattn** [comment]
@haosdent Thanks your review. I hope to add one more fix.

```
tensorflow/lite/c/c_api_internal.h:53:44: warning: 'struct TfLiteContext' declared inside parameter lis
t will not be visible outside of this definition or declaration
 typedef TfLiteStatus (*refreshFunc)(struct TfLiteContext* context);
   
diff --git a/tensorflow/lite/c/c_api_internal.h b/tensorflow/lite/c/c_api_internal.h
index 80d99d92e9..5939607d3c 100644
--- a/tensorflow/lite/c/c_api_internal.h
+++ b/tensorflow/lite/c/c_api_internal.h
@@ -50,6 +50,8 @@ typedef enum {
   kTfLiteMaxExternalContexts = 3
 } TfLiteExternalContextType;
 
+struct TfLiteContext;
+
 // An external context is a collection of information unrelated to the TF Lite
 // framework, but useful to a subset of the ops. TF Lite knows very little
 // about about the actual contexts, but it keeps a list of them, and is able to
```

Should be better to separate PR? or Can i include the patch into this PR?

**Review by haozha111 — DISMISSED** [review]

**Comment by haozha111** [comment]
> @haosdent Thanks your review. I hope to add one more fix.
> 
> ```
> tensorflow/tensorflow/tensorflow/lite/c/c_api_internal.h:60:34: warning: 'struct TfLiteContext' declared inside parameter lis
> t will not be visible outside of this definition or declaration
>    TfLiteStatus (*Refresh)(struct TfLiteContext* context);
> ```
> ```
> diff --git a/tensorflow/lite/c/c_api_internal.h b/tensorflow/lite/c/c_api_internal.h
> index 80d99d92e9..5939607d3c 100644
> --- a/tensorflow/lite/c/c_api_internal.h
> +++ b/tensorflow/lite/c/c_api_internal.h
> @@ -50,6 +50,8 @@ typedef enum {
>    kTfLiteMaxExternalContexts = 3
>  } TfLiteExternalContextType;
>  
> +struct TfLiteContext;
> +
>  // An external context is a collection of information unrelated to the TF Lite
>  // framework, but useful to a subset of the ops. TF Lite knows very little
>  // about about the actual contexts, but it keeps a list of them, and is able to
> ```
> Should be better to separate PR? or Can i include the patch into this PR?

Thanks for catching this! You can just make that change in this PR I think.

**Review by haozha111 — APPROVED** [review]

**Comment by haozha111** [comment]
Just  an update:

We decide to use enum to define the kTfLiteNullBufferHandle, not Macro. I will make this change internally and hold back this PR. Thanks!

**Comment by mattn** [comment]
Issue of `struct TfLiteContext` still not fixed?

**Comment by haozha111** [comment]
> Issue of `struct TfLiteContext` still not fixed?

I have fixed it yesterday. Only thing different from your change is that I used enum to declare the kTfLiteNullBufferHandle rather than Macro. Please see it here:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/c/c_api_internal.h

**Comment by mattn** [comment]
Thank you.

**Pull request closed without merging**